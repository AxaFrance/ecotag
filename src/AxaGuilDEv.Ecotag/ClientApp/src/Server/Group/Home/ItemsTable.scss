.af-table {
  &__small {
    width: 25px;
  }
}