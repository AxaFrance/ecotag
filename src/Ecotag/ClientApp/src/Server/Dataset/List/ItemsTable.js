import React from "react";
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import Loader from '@axa-fr/react-toolkit-loader';
import Action from "@axa-fr/react-toolkit-action";
import {useHistory} from "react-router-dom";
import {formatTimestampToString} from "../../date";
import {Locked} from "../Dataset.service";
import useProjectTranslation from "../../../translations/useProjectTranslation";

const getIconClassname = (locked) => {
    switch (locked) {
        case Locked.Locked:
            return 'btn af-btn--circle af-btn--locked';
        case Locked.LockedAndWorkInProgress:
            return 'btn af-btn--circle af-btn--locked-and-work-in-progress';
        case Locked.Pending:
            return 'btn af-btn--circle af-btn--pending';
        case Locked.None:
            return 'btn af-btn--circle af-btn--none';

    }
}

const getIcon = (locked) => {
    switch (locked) {
        case Locked.Locked:
        case Locked.LockedAndWorkInProgress:
            return 'lock';
        case Locked.Pending:
            return 'refresh';
        case Locked.None:
            return 'unlock';

    }
}

const ItemsTable = ({items, filters, loaderMode, onChangePaging}) => {

    const history = useHistory();
    const {translate} = useProjectTranslation();

    const editDatasetButton = id => {
        const path = `datasets/${id}`
        history.push(path);
    };

    const getTitle = (locked) => {
        switch (locked) {
            case Locked.Locked:
                return translate('dataset.list.actions_titles.locked');
            case Locked.LockedAndWorkInProgress:
                return translate('dataset.list.actions_titles.locked_and_work_in_progress');
            case Locked.Pending:
                return translate('dataset.list.actions_titles.pending');
            case Locked.None:
                return translate('dataset.list.actions_titles.none');
        }
    };

    return (
        <Loader mode={loaderMode}>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.lock')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.name')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.team')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.classification')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.nb_files')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.type')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.creation_date')}</span>
                        </Table.Th>
                        <Table.Th><span className="af-table__th-content">{translate('dataset.list.header.action')}</span></Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(
                        ({id, name, type, groupName, classification, numberFiles, createDate, locked}) => (
                            <Table.Tr key={id}>
                                <Table.Td>
                                    <Action className={getIconClassname(locked)}
                                            id="lock" icon={getIcon(locked)}
                                            title={getTitle(locked)}
                                            onClick={() => {
                                            }}/>
                                </Table.Td>
                                <Table.Td>{name}</Table.Td>
                                <Table.Td>{groupName}</Table.Td>
                                <Table.Td>{classification}</Table.Td>
                                <Table.Td>{numberFiles}</Table.Td>
                                <Table.Td>{type}</Table.Td>
                                <Table.Td>{formatTimestampToString(createDate)}</Table.Td>
                                <Table.Td>
                                    <Action id="id" icon="edit" title={translate('dataset.list.edit')} onClick={() => {
                                        editDatasetButton(id)
                                    }}/>
                                </Table.Td>
                            </Table.Tr>
                        )
                    )}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
                displayLabel={translate('paging.display')}
                elementsLabel={translate('paging.elements')}
                previousLabel={translate('paging.previous')}
                nextLabel={translate('paging.next')}
            />
        </Loader>

    )
};

export default ItemsTable;