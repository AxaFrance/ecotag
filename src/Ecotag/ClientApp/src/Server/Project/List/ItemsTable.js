import HeaderColumnCell from './ColumnHeader';
import React, {useEffect, useState} from 'react';
import {useHistory} from 'react-router-dom';
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import Action from '@axa-fr/react-toolkit-action';
import {formatTimestampToString} from '../../date';
import {fetchAnnotationsStatus} from '../Project.service';
import {resilienceStatus} from '../../shared/Resilience';
import {Popover} from '@axa-fr/react-toolkit-all';
import {ProgressBar} from './ProgressBar';
import useProjectTranslation from '../../../translations/useProjectTranslation';

const NumberTagToDo = ({state}) => {
    const {translate} = useProjectTranslation();
    const {ERROR, SUCCESS, LOADING} = resilienceStatus;
    const annotationsStatus = state.annotationsStatus;
    return <>{{
        [LOADING]: <span>{translate('project.projects_list.list.nb_tags_to_do.resilience.loading')}</span>,
        [ERROR]: (
            <span>{translate('project.projects_list.list.nb_tags_to_do.resilience.error')}</span>
        ),
        [SUCCESS]: <>
            <div className="projects-af-popover__wrapper">
                <Popover
                    placement="top"
                    classModifier="project-home"
                >
                    <Popover.Pop>
                        <h3>{translate('project.projects_list.list.nb_tags_to_do.title')}</h3>
                        <table>
                            <tr>
                                <td>{translate('project.projects_list.list.nb_tags_to_do.to_do')}</td>
                                <td>{translate('project.projects_list.list.nb_tags_to_do.done')}</td>
                                <td>{translate('project.projects_list.list.nb_tags_to_do.remaining')}</td>
                            </tr>
                            <tr>
                                <td>{annotationsStatus.numberAnnotationsToDo}</td>
                                <td>{annotationsStatus.numberAnnotationsDone}</td>
                                <td>{annotationsStatus.numberAnnotationsToDo - annotationsStatus.numberAnnotationsDone}</td>
                            </tr>
                        </table>
                    </Popover.Pop>
                    <Popover.Over>
                        <ProgressBar percentage={annotationsStatus.percentageNumberAnnotationsDone}
                                     label={annotationsStatus.percentageNumberAnnotationsDone === 100 ?
                                         translate('project.projects_list.list.nb_tags_to_do.percentage.done') :
                                         translate('project.projects_list.list.nb_tags_to_do.percentage.pending')}
                        />
                    </Popover.Over>
                </Popover>
            </div>
        </>
    }[state.status]
    }
    </>

}


const initAsync = (fetch) => async (state, setState, id) => {

    const annotationsStatusResponse = await fetchAnnotationsStatus(fetch)(id);
    if (annotationsStatusResponse.status >= 500 || annotationsStatusResponse.status === 403) {
        const data = {
            ...state,
            status: annotationsStatusResponse.status === 403 ? resilienceStatus.FORBIDDEN : resilienceStatus.ERROR
        };
        setState(data);
    } else {
        const annotationsStatus = await annotationsStatusResponse.json();
        const data = {annotationsStatus, status: resilienceStatus.SUCCESS};
        setState(data);
    }

}

const ProjectRow = ({id, name, groupName, createDate, annotationType, fetch}) => {
    const history = useHistory();
    const {translate} = useProjectTranslation();
    const projectPageButton = id => {
        const path = `/projects/${id}`;
        history.push(path);
    };
    const [state, setState] = useState({
        annotationsStatus: {
            isAnnotationClosed: true,
            numberAnnotationsByUsers: [],
            numberAnnotationsDone: 0,
            numberAnnotationsToDo: 0,
            percentageNumberAnnotationsDone: 0
        },
        status: resilienceStatus.LOADING
    })
    useEffect(async () => {
        initAsync(fetch)(state, setState, id);
    }, []);

    return (
        <Table.Tr key={id}>
            <Table.Td>{name}</Table.Td>
            <Table.Td>{groupName}</Table.Td>
            <Table.Td>{createDate}</Table.Td>
            <Table.Td>{annotationType}</Table.Td>
            <Table.Td>
                <NumberTagToDo state={state}/>
            </Table.Td>
            <Table.Td>
                <Action id="id" icon="zoom-in" title={translate('project.projects_list.list.headers.edit')} onClick={() => projectPageButton(id)}/>
            </Table.Td>
        </Table.Tr>
    );
};

const ItemsTable = ({items, filters, onChangePaging, onChangeSort, fetch}) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('name')}
                            headerColumnName={translate('project.projects_list.list.headers.name')}
                            filterColumnValue={filters.columns.name.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('groupName')}
                            headerColumnName={translate('project.projects_list.list.headers.team')}
                            filterColumnValue={filters.columns.groupName.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('createDate')}
                            headerColumnName={translate('project.projects_list.list.headers.creation_date')}
                            filterColumnValue={filters.columns.createDate.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('annotationType')}
                            headerColumnName={translate('project.projects_list.list.headers.annotation_type')}
                            filterColumnValue={filters.columns.annotationType.value}
                        />
                        <HeaderColumnCell
                            headerColumnName={translate('project.projects_list.list.headers.status')}
                        />
                        <Table.Th classModifier="sortable">
                            <span className="af-table__th-content af-btn__text">{translate('project.projects_list.list.headers.action')}</span>
                        </Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(({id, name, groupName, createDate, numberTagToDo, annotationType}) => (
                        <ProjectRow
                            key={id}
                            id={id}
                            name={name}
                            groupName={groupName}
                            createDate={formatTimestampToString(createDate)}
                            numberTagToDo={numberTagToDo}
                            annotationType={annotationType}
                            fetch={fetch}
                        />
                    ))}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
                displayLabel={translate('paging.display')}
                elementsLabel={translate('paging.elements')}
                previousLabel={translate('paging.previous')}
                nextLabel={translate('paging.next')}
            />
        </>
    )

};

export default ItemsTable;