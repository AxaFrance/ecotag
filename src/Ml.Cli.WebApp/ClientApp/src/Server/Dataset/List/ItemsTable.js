import React from "react";
import Table, { Paging } from '@axa-fr/react-toolkit-table';
import Loader from '@axa-fr/react-toolkit-loader';
import HeaderColumnCell from "../../Project/List/ColumnHeader";
import Action from "@axa-fr/react-toolkit-action";
import {useHistory} from "react-router-dom";
import {formatTimestampToString} from "../../date";
import {Locked} from "../Dataset.service";

const getIconClassname = (locked) =>{
    switch (locked) {
        case Locked.Locked:
            return 'btn af-btn--circle af-btn--locked';
        case Locked.Pending:
            return 'btn af-btn--circle af-btn--locked';
        case Locked.None:
            return 'btn af-btn--circle';
            
    }
}

const getIcon = (locked) =>{
    switch (locked) {
        case Locked.Locked:
            return 'lock';
        case Locked.Pending:
            return 'refresh';
        case Locked.None:
            return 'unlock';

    }
}

const getTitle = (locked) =>{
    switch (locked) {
        case Locked.Locked:
            return 'vérouillée';
        case Locked.Pending:
            return 'en cours';
        case Locked.None:
            return 'dévérouillée';

    }
}

const ItemsTable = ({items, filters, loaderMode, onChangePaging}) => {

    const history = useHistory();
    
    const editDatasetButton = id => {
        const path = `datasets/${id}`
        history.push(path);
    };
    
    return(
            <Loader mode={loaderMode}>
                <Table>
                    <Table.Header>
                        <Table.Tr>
                            <Table.Th>
                                <span className="af-table__th-content">Verrouiller</span>
                            </Table.Th>
                            <Table.Th>
                                <span className="af-table__th-content">Nom</span>
                            </Table.Th>
                            <Table.Th>
                                <span className="af-table__th-content">Groupe</span>
                            </Table.Th>
                            <Table.Th>
                                <span className="af-table__th-content">Classification</span>
                            </Table.Th>
                            <Table.Th>
                                <span className="af-table__th-content">Nombre de fichiers</span>
                            </Table.Th>
                            <Table.Th>
                                <span className="af-table__th-content">Type</span>
                            </Table.Th>
                            <Table.Th>
                                <span className="af-table__th-content">Date création</span>
                            </Table.Th>
                            <Table.Th><span className="af-table__th-content">Action</span></Table.Th>
                        </Table.Tr>
                    </Table.Header>
                    <Table.Body>
                        {items.map(
                            ({id, name, type, groupName, classification, numberFiles, createDate, locked}) => (
                                <Table.Tr key={id}>
                                    <Table.Td>
                                        <Action className={getIconClassname(locked)}
                                                id="lock" icon={getIcon(locked)}
                                                title={getTitle(locked)}
                                                onClick={() => {}} />
                                    </Table.Td>
                                    <Table.Td>{name}</Table.Td>
                                    <Table.Td>{groupName}</Table.Td>
                                    <Table.Td>{classification}</Table.Td>
                                    <Table.Td>{numberFiles}</Table.Td>
                                    <Table.Td>{type}</Table.Td>
                                    <Table.Td>{formatTimestampToString(createDate)}</Table.Td>
                                    <Table.Td>
                                        <Action id="id" icon="edit" title="Editer" onClick={() => {editDatasetButton(id)}} />
                                    </Table.Td>
                                </Table.Tr>
                            )
                        )}
                    </Table.Body>
                </Table>
                <Paging
                    onChange={onChangePaging}
                    numberItems={filters.paging.numberItemsByPage}
                    numberPages={filters.paging.numberPages}
                    currentPage={filters.paging.currentPage}
                    id="home_paging"
                />
            </Loader>

    )
};

export default ItemsTable;