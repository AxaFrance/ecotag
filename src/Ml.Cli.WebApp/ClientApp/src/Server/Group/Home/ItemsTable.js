import React, {useState} from "react";
import Table, { Paging } from '@axa-fr/react-toolkit-table';
import Action from '@axa-fr/react-toolkit-action';
import BooleanModal from '@axa-fr/react-toolkit-modal-boolean';
import Edit from "./Edit";

const DeleteGroupModal = ({ idGroup, isDeleteModalVisible, setDeleteModalVisible, onDeleteGroup }) => {
    return (
        <BooleanModal
            className={'af-modal'}
            isOpen={isDeleteModalVisible}
            title={'Confirmer la suppression de groupe ?'}
            id={'deleteModalId'}
            onCancel={() => setDeleteModalVisible(false)}
            onOutsideTap={() => setDeleteModalVisible(false)}
            onSubmit={() => {
                onDeleteGroup(idGroup);
                setDeleteModalVisible(false);
                return false;
            }}
            submitTitle={'Supprimer'}
            cancelTitle={'Annuler'}>
            <p>Confirmez-vous la suppression de ce groupe ?</p>
        </BooleanModal>
    );
};

const UserRow = ({ id, name, users, eligibleUsers, onUpdateUser, onDeleteGroup }) => {
    const [isDeleteModalVisible, setDeleteModalVisible] = useState(false);
    const [isManageUsersModalVisible, setManageUsersModalVisible] = useState(false);
    return (
        <Table.Tr key={id}>
            <Table.Td>{name}</Table.Td>
            <Table.Td>{users.map(user => user.email).join(', ')}</Table.Td>
            <Table.Td>
                <Action id="editActionId" icon="pencil" title="Modifier" onClick={() => setManageUsersModalVisible(true)} />
                <Edit
                    idGroup={id}
                    users={users}
                    eligibleUsers={eligibleUsers}
                    onUpdateUser={onUpdateUser}
                    isManageUsersModalVisible={isManageUsersModalVisible}
                    setManageUsersModalVisible={setManageUsersModalVisible}
                />
                <Action id="removeActionId" icon="remove" title="Supprimer" onClick={() => setDeleteModalVisible(true)} />
                <DeleteGroupModal
                    idGroup={id}
                    isDeleteModalVisible={isDeleteModalVisible}
                    setDeleteModalVisible={setDeleteModalVisible}
                    onDeleteGroup={onDeleteGroup}
                />
            </Table.Td>
        </Table.Tr>
    );
};

const ItemsTable = ({items, filters, onChangePaging, onDeleteGroup, onUpdateUser}) => {
    console.log(items);
    return(
        <>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <Table.Th>
                            <span className="af-table__th-content">Nom du groupe</span>
                        </Table.Th>
                        <Table.Th>Utilisateurs</Table.Th>
                        <Table.Th>Actions</Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(({ id, name, users = [], eligibleUsers = [] }) => (
                        <UserRow
                            key={id}
                            id={id}
                            name={name}
                            users={users}
                            eligibleUsers={eligibleUsers}
                            onUpdateUser={onUpdateUser}
                            onDeleteGroup={onDeleteGroup}
                        />
                    ))}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
            />
        </>
    )
};

export default ItemsTable;