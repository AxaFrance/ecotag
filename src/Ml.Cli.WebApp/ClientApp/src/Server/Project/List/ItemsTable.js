import HeaderColumnCell from "./ColumnHeader";
import React, {useEffect, useState} from "react";
import {useHistory} from "react-router-dom";
import Table, { Paging } from '@axa-fr/react-toolkit-table';
import Action from '@axa-fr/react-toolkit-action';
import {formatTimestampToString} from "../../date";
import {fetchAnnotationsStatus} from "../Project.service";
import {resilienceStatus} from "../../shared/Resilience";


const NumberTagToDo = ({state}) =>{
    const {ERROR, SUCCESS, LOADING} = resilienceStatus;
    const annotationsStatus = state.annotationsStatus;
    return <>{{
        [LOADING]: <span>Chargement...</span>,
        [ERROR]: (
            <span>Erreur...</span>
        ),
        [SUCCESS]: <>
            <progress value={annotationsStatus.percentageNumberAnnotationsDone} max="100"/>
            <span>{annotationsStatus.numberAnnotationsDone+ " / " + annotationsStatus.numberAnnotationsToDo}</span>
        </>
    }[state.status]
    }
    </>
    
}


const ProjectRow = ({ id, name, groupName, createDate, annotationType, fetch }) => {
    const history = useHistory();
    const projectPageButton = id => {
        const path = `/projects/${id}`;
        history.push(path);
    };
    const [state, setState] = useState({
        annotationsStatus: {
                isAnnotationClosed: true,
                numberAnnotationsByUsers: [],
                numberAnnotationsDone: 0,
                numberAnnotationsToDo:0,
                percentageNumberAnnotationsDone:0
        },
        status: resilienceStatus.LOADING
    })
    useEffect(async () => {

        const annotationsStatusResponse = await fetchAnnotationsStatus(fetch)(id);
        if (annotationsStatusResponse.status >= 500) {
            const data = {
                ...state,
                status: resilienceStatus.ERROR
            };
            setState(data);
        } else {
            const annotationsStatus = await annotationsStatusResponse.json();
            const data = {annotationsStatus, status: resilienceStatus.SUCCESS};
            setState(data);
        }

    }, []);
    
    return (
        <Table.Tr key={id}>
            <Table.Td>{name}</Table.Td>
            <Table.Td>{groupName}</Table.Td>
            <Table.Td>{createDate}</Table.Td>
            <Table.Td>{annotationType}</Table.Td>
            <Table.Td><NumberTagToDo state={state}/></Table.Td>
            <Table.Td>
                <Action id="id" icon="zoom-in" title="Editer" onClick={() => projectPageButton(id)} />
            </Table.Td>
        </Table.Tr>
    );
};

const ItemsTable = ({items, filters, onChangePaging, onChangeSort, fetch}) => {
    return(
        <>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('name')}
                            headerColumnName={'Nom'}
                            filterColumnValue={filters.columns.name.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('groupName')}
                            headerColumnName={'Groupe'}
                            filterColumnValue={filters.columns.groupName.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('createDate')}
                            headerColumnName={'Date création'}
                            filterColumnValue={filters.columns.createDate.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('typeAnnotation')}
                            headerColumnName={"Type d'annotation"}
                            filterColumnValue={filters.columns.typeAnnotation.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('numberTagToDo')}
                            headerColumnName={'Tags restants'}
                            filterColumnValue={filters.columns.numberCrossAnnotation.value}
                        />
                        <Table.Th classModifier="sortable">
                            <span className="af-table__th-content af-btn__text">Action</span>
                        </Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(({ id, name, groupName, createDate, numberTagToDo, annotationType }) => (
                        <ProjectRow
                            key={id}
                            id={id}
                            name={name}
                            groupName={groupName}
                            createDate={formatTimestampToString(createDate)}
                            numberTagToDo={numberTagToDo}
                            annotationType={annotationType}
                            fetch={fetch}
                        />
                    ))}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
            />
        </>
    )
    
};

export default ItemsTable;