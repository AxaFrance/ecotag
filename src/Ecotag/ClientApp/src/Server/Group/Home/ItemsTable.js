import React, {useState} from "react";
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import Action from '@axa-fr/react-toolkit-action';
import Edit from "./Edit";
import './ItemsTable.scss';
import useProjectTranslation from "../../../useProjectTranslation";

const computeUsersList = users => {
    if (users.length === 0) {
        return "-";
    }
    return users.map(user => user.email).join(', ');
};

const UserRow = ({id, name, users, eligibleUsers, onUpdateUser}) => {
    const [isManageUsersModalVisible, setManageUsersModalVisible] = useState(false);
    const {translate} = useProjectTranslation();
    return (
        <Table.Tr key={id}>
            <Table.Td>{name}</Table.Td>
            <Table.Td>{computeUsersList(users)}</Table.Td>
            <Table.Td class="af-table__cell af-table__small">
                <Action id="editActionId" icon="pencil" title={translate('group.list.edit')}
                        onClick={() => setManageUsersModalVisible(true)}/>
                <Edit
                    idGroup={id}
                    users={users}
                    eligibleUsers={eligibleUsers}
                    onUpdateUser={onUpdateUser}
                    isManageUsersModalVisible={isManageUsersModalVisible}
                    setManageUsersModalVisible={setManageUsersModalVisible}
                />
            </Table.Td>
        </Table.Tr>
    );
};

const ItemsTable = ({items, filters, onChangePaging, onUpdateUser}) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.name')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.users')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.actions')}</span>
                        </Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(({id, name, users = [], eligibleUsers = []}) => (
                        <UserRow
                            key={id}
                            id={id}
                            name={name}
                            users={users}
                            eligibleUsers={eligibleUsers}
                            onUpdateUser={onUpdateUser}
                        />
                    ))}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
                displayLabel={translate('group.paging.display')}
                elementsLabel={translate('group.paging.elements')}
                previousLabel={translate('group.paging.previous')}
                nextLabel={translate('group.paging.next')}
            />
        </>
    )
};

export default ItemsTable;