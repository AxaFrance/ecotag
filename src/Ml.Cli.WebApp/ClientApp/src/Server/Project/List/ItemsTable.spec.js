import {render} from "@testing-library/react";
import {LoaderModes} from "@axa-fr/react-toolkit-all";
import ItemsTable from "./ItemsTable";

const items = [{
    id: "0001",
    name: "Carte verte",
    groupName: "groupName",
    annotationType: "Image",
    numberTagToDo: 300,
    createDate: new Date("10-30-2019").getTime()
}];

const filters = {
    paging: {
        numberItemsByPage: 10,
            currentPage: 1,
    },
    filterValue: null,
        columns: {
        name: { value: null, timeLastUpdate: null },
        groupName: {value: null, timeLastUpdate: null},
        createDate: { value: 'desc', timeLastUpdate: new Date() },
        annotationType: { value: null, timeLastUpdate: null },
        numberTagToDo: { value: null, timeLastUpdate: null },
        numberCrossAnnotation: {value: null}
    },
};

export const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));

export const fetch = async (url, config) => {
    await sleep(1);
            return {
                status: 201,
                json: async () => {
                    await sleep(1);
                    return {};
                }
            };
};

describe("Check Project ItemsTable behaviour", () => {
    
    test("Chould render correctly", async () => {
        const {asFragment} = render(<ItemsTable items={items} filters={filters} loaderMode={LoaderModes.none} onChangePaging={() => {}} onChangeSort={() => {}} fetch={fetch}/>);
        expect(asFragment()).toMatchSnapshot();
    })
    
})