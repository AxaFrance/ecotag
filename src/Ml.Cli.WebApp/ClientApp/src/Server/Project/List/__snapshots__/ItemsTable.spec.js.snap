// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Project ItemsTable behaviour Chould render correctly and display deletion modal 1`] = `
<DocumentFragment>
  <table
    class="af-table"
  >
    <thead
      class="af-table-thead"
    >
      <tr
        class="af-table__tr"
      >
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Nom
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Classification
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Date création
              </span>
              <i
                class="glyphicon glyphicon-sorting-desc"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Type d'annotation
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Tags restants
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content af-btn__text"
          >
            Action
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      class="af-table__body"
    >
      <tr
        class="af-table__tr"
      >
        <td
          class="af-table__cell"
        >
          Carte verte
        </td>
        <td
          class="af-table__cell"
        >
          Publique
        </td>
        <td
          class="af-table__cell"
        >
          -
        </td>
        <td
          class="af-table__cell"
        />
        <td
          class="af-table__cell"
        />
        <td
          class="af-table__cell"
        >
          <a
            class="btn af-btn--circle"
            href="#"
            id="id"
            role="button"
            tabindex="0"
            title="Editer"
          >
            <i
              class="glyphicon glyphicon-zoom-in"
            />
          </a>
          <a
            class="btn af-btn--circle"
            href="#"
            id="removeActionId"
            role="button"
            tabindex="0"
            title="Supprimer"
          >
            <i
              class="glyphicon glyphicon-remove"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="af-paging"
  >
    <div
      class="af-paging__limit"
    >
      <div
        class="af-paging__limit"
      >
        <form
          class="af-paging__form"
        >
          <div
            class="af-form__group"
          >
            <div
              class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
            >
              <label
                class="af-form__group-label"
                for="home_paging"
              >
                Afficher
              </label>
            </div>
            <div
              class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
            >
              <div
                class="af-form__select"
              >
                <div
                  class="af-form__select-container"
                >
                  <select
                    class="af-form__input-select"
                    id="home_paging"
                  >
                    <option
                      value="5"
                    >
                      5
                    </option>
                    <option
                      value="10"
                    >
                      10
                    </option>
                    <option
                      value="25"
                    >
                      25
                    </option>
                    <option
                      value="50"
                    >
                      50
                    </option>
                    <option
                      value="100"
                    >
                      100
                    </option>
                  </select>
                  <span
                    class="glyphicon glyphicon-menu-down"
                  />
                </div>
                <span
                  class="af-form__input-cmplt"
                >
                  éléments
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="af-paging__pager"
    >
      <nav
        class="af-pager"
      >
        <ul
          class="af-pager__pagination"
        >
          <li
            class="af-pager__item af-pager__item--disabled"
          >
            <span
              class="af-pager__item-nolink"
            >
              « Précédent
            </span>
          </li>
          <li
            class="af-pager__item af-pager__item--active"
          >
            <a
              class="af-pager__item-link"
              href="#"
            >
              <span>
                1
              </span>
            </a>
          </li>
          <li
            class="af-pager__item af-pager__item--disabled"
          >
            <span
              class="af-pager__item-nolink"
            >
              Suivant »
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
